### PR TITLE
CM_Cli_Exception_InvalidArguments is not catched by Cli manager

### PR DESCRIPTION
**Pull request by alexispeter**
```
$ bin/cm s3export create-job
CM_Cli_Exception_InvalidArguments: Missing argument `device-path` in /Users/tomasz/projects/cargomedia/s3export_backup/vendor/cargomedia/cm/library/CM/Cli/Arguments.php on line 119
    0. {main} bin/cm:0
    1. CM_Cli_CommandManager->run(CM_Cli_Arguments) /Users/tomasz/projects/cargomedia/s3export_backup/vendor/cargomedia/cm/bin/cm:18
    2. CM_Process->fork(Closure) /Users/tomasz/projects/cargomedia/s3export_backup/vendor/cargomedia/cm/library/CM/Cli/CommandManager.php:184
    3. CM_Process->_fork(Closure, 1) /Users/tomasz/projects/cargomedia/s3export_backup/vendor/cargomedia/cm/library/CM/Process.php:34
    4. CM_Process_ForkHandler->runAndSendWorkload() /Users/tomasz/projects/cargomedia/s3export_backup/vendor/cargomedia/cm/library/CM/Process.php:191
    5. CM_Cli_CommandManager->{closure}() /Users/tomasz/projects/cargomedia/s3export_backup/vendor/cargomedia/cm/library/CM/Process/ForkHandler.php:58
    6. CM_Cli_Command->run(CM_Cli_Arguments, CM_InputStream_Readline, CM_OutputStream_Stream_StandardOutput, CM_OutputStream_Stream_StandardError) /Users/tomasz/projects/cargomedia/s3export_backup/vendor/cargomedia/cm/library/CM/Cli/CommandManager.php:178
    7. CM_Cli_Arguments->extractMethodParameters(ReflectionMethod) /Users/tomasz/projects/cargomedia/s3export_backup/vendor/cargomedia/cm/library/CM/Cli/Command.php:27
    8. CM_Cli_Arguments->_getParamValue(ReflectionParameter) /Users/tomasz/projects/cargomedia/s3export_backup/vendor/cargomedia/cm/library/CM/Cli/Arguments.php:56
```

@njam I think this started to happen after adding forking mechanism.

### CODE REVIEW

**Comment by alexispeter** [comment]
please review @tomaszdurka

**Comment by tomaszdurka** [comment]
Can you make the some by catching CM_Cli_Exception_InvalidArguments inside that workload? I think it might be useful to throw CM_Cli_Exception_InvalidArguments from inside of the runnable/command.

**Comment by tomaszdurka** [comment]
Additionally it is a bit strange that `checkUnused` is on the arguments object, maybe let's move it to command-manager? Not sure about the whole structure.

**Comment by alexispeter** [comment]
> Can you make the some by catching CM_Cli_Exception_InvalidArguments inside that workload? I think it might be useful to throw CM_Cli_Exception_InvalidArguments from inside of the runnable/command.

This would require some major refactoring. The exception would need to be caught inside the executing process but still somehow propagated to the parent process so it can set the correct exit code. We could pass along the workloadresult to the workload and add some methods to manipulate it, similar to ajax-callbacks and ajax-response. Wdyt?

**Comment by alexispeter** [comment]
please review @tomaszdurka

File: library/CM/Process/WorkloadResult.php [hunk 1]
```diff
@@ -35,6 +32,15 @@ public function getException() {
     }
 
     /**
+     * @param Exception $exception
+     * @return $this
```
Comment (mariansollmann) [inline library/CM/Process/WorkloadResult.php]: sorry for jumping in. Just for me to know: Is `@return $this` correct? Why you use it and why should it not be `CM_Process_WorkloadResult `?

Comment (alexispeter) [inline library/CM/Process/WorkloadResult.php]: PhpStorm auto-complete. I guess it saves you from having to overwrite a setter just to get the current return-type in a inheritance-scenario.

Comment (njam) [inline library/CM/Process/WorkloadResult.php]: We usually do `@return static` afaik

Comment (alexispeter) [inline library/CM/Process/WorkloadResult.php]: I guess both works

**Comment by tomaszdurka** [comment]
Looks ok. 
- Rename `_getProcessWorkLoad` -> `_getProcessWorkload`

**Pull request merged**